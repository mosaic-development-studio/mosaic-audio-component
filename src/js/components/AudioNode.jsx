import PropTypes from 'prop-types';
import React from 'react';

import { REFS } from '../lib/constants';

export const AudioNode = ({
    audioNodeId,
    createRef,
    loop,
    movePlayHead,
    preload,
    source,
    updateDuration,
    updateTime
}) => {
    return (
        <audio
            id={audioNodeId}
            loop={loop}
            onDurationChange={updateDuration}
            onTimeUpdate={() => {
                updateTime();
                movePlayHead();
            }}
            preload={preload}
            ref={node => createRef(REFS.AUDIO, node)}
        >
            <source src={source} />
        </audio>
    );
};

AudioNode.defaultProps = {
    audioNodeId: null,
    loop: false,
    preload: 'auto'
};

AudioNode.propTypes = {
    audioNodeId: PropTypes.string,
    createRef: PropTypes.func.isRequired,
    loop: PropTypes.bool,
    movePlayHead: PropTypes.func.isRequired,
    preload: PropTypes.string,
    source: PropTypes.string.isRequired,
    updateDuration: PropTypes.func.isRequired,
    updateTime: PropTypes.func.isRequired
};

export default AudioNode;
