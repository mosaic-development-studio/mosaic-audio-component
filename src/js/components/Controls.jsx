import PropTypes from 'prop-types';
import React from 'react';

export const Controls = ({ children }) => {
    return (
        <div className="riverine-controls">
            {children}
        </div>
    );
};

Controls.propTypes = {
    children: PropTypes.node.isRequired
};

export default Controls;
