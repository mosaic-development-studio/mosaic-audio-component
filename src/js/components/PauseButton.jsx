import PropTypes from 'prop-types';
import React from 'react';

export const PauseButton = ({ pause }) => {
    return (
        <li className="pause-button-container">
            <a className="riverine-pause" onClick={pause}>
                <i className="fa fa-pause"></i>
            </a>
        </li>
    );
};

PauseButton.propTypes = {
    pause: PropTypes.func.isRequired
};

export default PauseButton;
