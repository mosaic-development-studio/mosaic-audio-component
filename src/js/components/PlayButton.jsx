import PropTypes from 'prop-types';
import React from 'react';

export const PlayButton = ({ play }) => {
    return (
        <li className="play-button-container">
            <a className="riverine-play" onClick={play}>
                <i className="fa fa-play"></i>
            </a>
        </li>
    );
};

PlayButton.propTypes = {
    play: PropTypes.func.isRequired
};

export default PlayButton;
