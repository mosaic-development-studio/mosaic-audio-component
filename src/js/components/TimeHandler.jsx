import PropTypes from 'prop-types';
import React from 'react';

export const TimeHandler = props => {
    return (
        <div className="riverine-time-handler">
            <span>{props.currentTime} / {props.duration}</span>
        </div>
    );
};

TimeHandler.propTypes = {
    currentTime: PropTypes.string.isRequired,
    duration: PropTypes.string.isRequired
};

export default TimeHandler;