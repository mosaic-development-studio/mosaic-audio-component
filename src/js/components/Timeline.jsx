import PropTypes from 'prop-types';
import React from 'react';

import { REFS } from '../lib/constants';

export const Timeline = ({
    createRef,
    handleMouseDown,
    handleMouseMove,
    handleMouseOut,
    hoverWidth,
    playHeadPaddingLeft
}) => {
    return (
        <div className="riverine-progress"
            onMouseDown={handleMouseDown}
            onMouseMove={handleMouseMove}
            onMouseOut={handleMouseOut}
            ref={node => createRef(REFS.TIMELINE, node)}
        >
            <div className="riverine-seek-bar" style={{ width: hoverWidth }}>
                <div
                    className="riverine-play-bar"
                    onMouseDown={handleMouseDown}
                    ref={node => createRef(REFS.PLAYHEAD, node)}
                    style={{
                        paddingLeft: playHeadPaddingLeft
                    }}
                />
            </div>
        </div>
    );
};

Timeline.propTypes = {
    createRef: PropTypes.func.isRequired,
    handleMouseDown: PropTypes.func.isRequired,
    handleMouseMove: PropTypes.func.isRequired,
    handleMouseOut: PropTypes.func.isRequired,
    hoverWidth: PropTypes.string.isRequired,
    playHeadPaddingLeft: PropTypes.string.isRequired
};

export default Timeline;
