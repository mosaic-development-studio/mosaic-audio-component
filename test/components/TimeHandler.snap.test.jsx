import React from 'react';
import renderer from 'react-test-renderer';

import { TimeHandler } from '../../src/js/components/TimeHandler';

describe('TimeHandler snapshots', () => {
    it('should match the snapshot with required props', () => {
        const TimeHandlerComponent = renderer
            .create(
                <TimeHandler
                    currentTime="00:00"
                    duration="02:32"
                />
            )
            .toJSON();

        expect(TimeHandlerComponent).toMatchSnapshot();
    });
});