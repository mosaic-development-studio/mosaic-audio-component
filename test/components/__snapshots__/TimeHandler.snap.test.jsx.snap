// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeHandler snapshots should match the snapshot with required props 1`] = `
<div
  className="riverine-time-handler"
>
  <span>
    00:00
     / 
    02:32
  </span>
</div>
`;
